components/tidb_query_expr/src/impl_string.rs

```diff
@@ -702,15 +702,15 @@ pub fn instr_utf8(s: BytesRef, substr: BytesRef) -> Result<Option<Int>> {
 
 #[rpn_fn]
 #[inline]
-pub fn find_in_set(s: BytesRef, str_list: BytesRef) -> Result<Option<Int>> {
+pub fn find_in_set<C: Collator>(s: BytesRef, str_list: BytesRef) -> Result<Option<Int>> {
     if str_list.is_empty() {
         return Ok(Some(0));
     }
 
-    let s = String::from_utf8_lossy(s);
+    use std::cmp::Ordering::*;
     let result = String::from_utf8_lossy(str_list)
         .split(',')
-        .position(|str_in_set| str_in_set == s)
+        .position(|str_in_set| C::sort_compare(str_in_set.as_bytes(), s).unwrap() == Equal)
```

Comment:
```suggestion
        .position(|str_in_set| C::sort_compare(str_in_set.as_bytes(), s).filter(|o| o == Ordering::Equal).is_some())
```

Reply:
After `C::sort_compare(str_in_set.as_bytes(), s)`, `filter` can't be used.

Reply:
Maybe use `ok()` to convert `Result<T>` to `Option<T>` before using `filter()`?